threedi_scenario_downloader/downloader.py

```diff
@@ -102,15 +102,21 @@ def get_scenario_instance(scenario_uuid):
     return scenario_instance
 
 
-def get_scenario_instance_results(scenario_uuid):
+def get_scenario_instance_results(scenario_uuid, subendpoint=None):
     """check if scenario exist"""
     get_scenario_instance(scenario_uuid)
 
+    if subendpoint:
+        url = "{}scenarios/{}/results/{}".format(LIZARD_URL, scenario_uuid, subendpoint)
+    else:
+        url = "{}scenarios/{}/results".format(LIZARD_URL, scenario_uuid)
+
     """return request response with all raster results"""
-    r = requests.get(
-        url="{}scenarios/{}/results".format(LIZARD_URL, scenario_uuid),
-        auth=("__key__", get_api_key()),
-    )
+    r = requests.get(url=url, auth=("__key__", get_api_key()))
+    if not r.json()["results"]:
+        logging.debug("The result data you request is non-existent")
+        raise ValueError("The result data you request is non-existent")
+
     r.raise_for_status()
```

Comment:
De `raise_for_status()` kan je beter gelijk na de `requests.get()` doen. Dat zijn meestal dingen die zorgen dat `r.json()` ook fout gaan => beter gelijk.

Reply:
De raise for status() is naar voren geplaatst.